aten/src/ATen/native/TensorFactories.cpp

```diff
@@ -131,7 +131,7 @@ Tensor eye(int64_t n, const TensorOptions& options) {
 }
 
 Tensor eye(int64_t n, int64_t m, const TensorOptions& options) {
-  auto tensor = options.type().tensor({});
+  auto tensor = at::getType(options).tensor({});
```

Comment:
I'd maybe create a utility/detail function at the top of this file called `raw_tensor` that takes an options and an `IntList` for the size and wraps this -- it's a bit repetitive right now. Maybe with a comment stating that `type().tensor()` is the lowest level interface that allocates a tensor without a device specified.

Reply:
Actually, I'm a little perplexed: why can't we do `at::tensor({}, options)` here? What's the difference?

Reply:
What you call `raw_tensor` is just `native::empty`. I'm going to call that instead.

Reply:
`at::tensor({}, options)` would work, but a `DeviceGuard` will have been set for all of these methods in e.g. `at::eye`, so `at::tensor` would set yet another DeviceGuard. That should technically never be a problem, but conceptually at this point a `DeviceGuard` has been set, so we can just go straight to allocating a raw tensor (which may also save some microseconds to avoid the duplicate `DeviceGuard` construction). `native::empty` sounds good to me!